lib/packages/screens/app.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:coffee_app/packages/authentication/bloc/authentication_bloc.dart';
+import 'package:coffee_app/packages/authenticationRepository/authentication_repository.dart';
+import 'package:coffee_app/packages/screens/splash_screen.dart';
+import 'package:coffee_app/packages/userRepository/user_repository.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+import '../../features/screens/home_screen.dart';
+import '../login/screen/login_screen.dart';
+
+class App extends StatefulWidget {
+   const App({super.key});
+
+   @override
+   State<App> createState() => _AppState();
+ }
+
+ class _AppState extends State<App> {
+
+  late final AuthenticationRepository _authenticationRepository;
+  late final UserRepository _userRepository;
+
+  @override
+  void initState() {
+    super.initState();
+    _authenticationRepository = AuthenticationRepository();
+    _userRepository = UserRepository();
+  }
+
+  @override
+  void dispose() {
+    _authenticationRepository.dispose();
```

Comment:
Not to complicate this whole thing with having repositories inside state, you could use 
https://pub.dev/documentation/provider/latest/provider/Provider-class.html

Then it will look like
```dart
Provider(
  create: () => AuthenticationRepository(),
  dispose: (repository) => repository.dispose(),
  child: BlocProvider(
  /// your bloc an other stuff
  )
)
```

Then you don't need to store any repository inside a state